lib/Activity.py

```diff
@@ -768,7 +775,7 @@ def resume(self):
             self.config.logger.error(f"Unable to get activity {self.name} from backend.")
             return
 
-        if last_activity["state"] == 'in_progress':
+        if last_activity["state"] == 'in_progress' and last_activity["status"].lower() not in ["succeeded", "failed"]:
```

Comment:
This might need future tweaking.  Not sure about which states indicate a status of done.